framework/model/module/module.go

```diff
@@ -83,6 +84,7 @@ type ModuleOptions struct {
 }
 
 type Module interface {
+	Init(opts ModuleOptions) error
```

Comment:
why we need `ModuleOptions` for both init and run/setup?

Reply:
also we should have comments to explain the difference of these two methods.

Reply:
only setup is needed

setup: init data struct and register runable functions